src/logic/map_objects/tribes/constructionsite.cc

```diff
@@ -384,6 +384,7 @@ void ConstructionSite::enhance(Game&) {
 	   uint32_t old_max, uint32_t old_des, uint32_t new_max) { return old_des * new_max / old_max; };
 
 	BuildingSettings* old_settings = settings_.release();
+	assert(old_settings);
```

Comment:
Why not make `BuildingSettings* old_settings` a `unique_ptr` too? This way, we can get rid of the ugly raw delete.

Reply:
Done